### ISSUE TITLE
D and Z values have opposite signs?

### ISSUE THREAD

**Issue by xuelingyi** [smussmann82/Comp-D_MPI#2]
Hello, in my analyses all three methods (d, p, f) gave results of D and Z in opposite values. In other words, positive D statistics with negative Z scores, and negative D with positive Z scores. Since the null expectation is D=0, I think Z and D should be of the same sign. Could you please explain why this is not the case? Thanks.

**Comment by stevemussmann** [comment]
Hello,

The function being used to calculate the Z score is in the Stats.cpp file:

`double Stats::calcZ(double D, double sd)
{
    double Z;
    if(sd==0)
    {
        Z=0.0;
    }
    else
    {
        Z = (0.0 - D) / sd;
    }
    return Z;
}`

The value of D ends up being positive or negative depending upon which taxon in the tree (P1 or P2) shows evidence of discordant sites as a consequence of mixing with P3. I probably should be taking the absolute value of Z here, but I do this later in the Dstat class when the P-value is calculated:

`boost::math::normal_distribution<> zdist(0.0, 1.0);
Zpval = 2.0*(1-boost::math::cdf(zdist, fabs(Z)));`

So basically, the sign of the Z value being the opposite of the D score is a printing error. Thanks for catching this - I will fix this when I next push revisions to this repository.

**Comment by xuelingyi** [comment]
Got it. Thanks for clarifying!
